scripts/tips.py

```diff
@@ -0,0 +1,217 @@
+
+
+mpl.rcParams['axes.linewidth'] = 1.5
+
+margin = 0.01
+rect = [margin, margin, 1-2*margin, 1-2*margin]
+
+
+# color range
+# -----------------------------------------------------------------------------
+fig = plt.figure(figsize=(2, 2))
+ax = fig.add_axes(rect, xticks=[], yticks=[])
+
+np.random.seed(1)
+X = np.random.randn(1000, 4)
+cmap = plt.get_cmap("Oranges")
+colors = [cmap(i) for i in [0.1, 0.3, 0.5, 0.7]]
+ax.hist(X, 2, density=True, histtype='bar', color=colors)
+
+fig.savefig("../figures/tip-color-range.pdf")
+
+
+# colorbar
+# -----------------------------------------------------------------------------
+fig = plt.figure(figsize=(2.15, 2))
+ax = fig.add_axes(rect, xticks=[], yticks=[])
+
+np.random.seed(1)
+Z = np.random.uniform(0, 1, (8, 8))
+cmap = plt.get_cmap("Oranges")
+im = ax.imshow(Z, interpolation="nearest", cmap=cmap, vmin=0, vmax=2)
+cb = fig.colorbar(im, fraction=0.046, pad=0.04)
+cb.set_ticks([])
+
+fig.savefig("../figures/tip-colorbar.pdf")
+
+
+# dotted
+# -----------------------------------------------------------------------------
+fig = plt.figure(figsize=(5, 0.25))
+
+ax = fig.add_axes(
+    [0, 0, 1, 1], frameon=False, xticks=[], yticks=[], xlim=[0, 1], ylim=[-0.5, 1.5]
+)
+
+epsilon=1e-12
+ax.plot([0, 1], [0, 0], "black", clip_on=False, lw=8,
+        ls=(.5, (epsilon, 1)), dash_capstyle="round")
+ax.plot([0, 1], [1, 1], "black", clip_on=False, lw=8,
+        ls=(-.5, (epsilon, 2)), dash_capstyle="round")
+fig.savefig("../figures/tip-dotted.pdf")
+
+
+# dual axis
+# -----------------------------------------------------------------------------
+fig = plt.figure(figsize=(2, 2))
+ax1 = fig.add_axes(rect, label="cartesian")
+ax2 = fig.add_axes(rect, projection="polar", label="polar")
+
+ax1.set_xticks([])  # np.linspace(0.0, 0.4, 5))
+ax1.set_yticks([])  # np.linspace(0.0, 1.0, 11))
+
+ax2.set_rorigin(0)
+ax2.set_thetamax(90)
+ax2.set_ylim(0.5, 1.0)
+ax2.set_xticks(np.linspace(0, np.pi/2, 10))
+ax2.set_yticks(np.linspace(0.5, 1.0, 5))
+
+ax2.set_xticklabels([])
+ax2.set_yticklabels([])
+
+fig.savefig("../figures/tip-dual-axis.pdf")
+
+
+# font family
+# -----------------------------------------------------------------------------
+# Setup a plot such that only the bottom spine is shown
+def setup(ax):
+    ax.spines['right'].set_color('none')
+    ax.spines['left'].set_color('none')
+    ax.yaxis.set_major_locator(mpl.ticker.NullLocator())
+    ax.spines['top'].set_color('none')
+
+    ax.spines['bottom'].set_linewidth(1)
+    ax.spines['bottom'].set_position("center")
+
+    ax.xaxis.set_ticks_position('bottom')
+    ax.tick_params(which='major', width=1.00)
+    ax.tick_params(which='major', length=5)
+    ax.tick_params(which='minor', width=0.75)
+    ax.tick_params(which='minor', length=2.5)
+    ax.set_xlim(0, 5)
+    ax.set_ylim(0, 1)
+    ax.patch.set_alpha(0.0)
+
+
+fig = plt.figure(figsize=(5, 0.5))
+fig.patch.set_alpha(0.0)
+n = 1
+
+fontsize = 18
+ax = plt.subplot(n, 1, 1)
```

Comment:
Join `fig`, and `ax`, I think.
```suggestion
fig, ax = plt.subplots(figsize=(5, 0.5))
fig.patch.set_alpha(0.0)

fontsize = 18
```